modules/basicfuncs/list-funcs.c

```diff
@@ -25,9 +25,9 @@
 #include "str-repr/encode.h"
 
 static void
-_append_comma(GString *result)
+_append_comma(GString *result, gsize initial_len)
 {
-  if (result->len == 0)
+  if (result->len == initial_len)
```

Comment:
From clean code perspective it is hard to understand what does `initial_len` do for an `_append_comma` function. Previously, when initial len was zero, that version was also strange but I could reason with something similar: I do not need to use comma when appending to an empty string. With this change, although the logic is very similar, still it is harder to reason about it.

Maybe this condition should be moved outside the `_append_comma`?

Reply:
I have renamed the function to better indicate that this is basically a private function, part of the various list functions and doing a bit more than just appending a comma. While I agree it is a bit cumbersome to follow the logic and why initial_len is important to the decision of appending a comma, but I would rather not do more than the minimal fix to the issue. The goal of this branch was to improve test coverage after all. I'd defer refactoring of the list functions to a later date.

I've also added a comment to explain the reasons for that check.